src/plugin.ts

```diff
@@ -173,5 +201,14 @@ function serializeResource(resource: ast.Resource): string {
  * does not support more than 1 element in a pattern.
  */
 function serializeMessage(message: ast.Message): [id: string, value: string] {
-  return [message.id.name, (message.pattern.elements[0] as ast.Text).value];
-}
+  const newStringArr = [];
+  for( const element of message.pattern.elements) {
+    if (element.type === "Text") {
+      newStringArr.push(element.value);
+    } else if (element.type === "Placeholder") {
+      newStringArr.push(`{${element.body.name}}`);
+    }
```

Comment:
The plugin will crash on serialization if the variableReferencePattern is not `{}`. The pattern is hardcoded here. 

1. I recommend adding a test with another placeholder pattern like `[]`. 
2. Update this code to be non-hardcoded. 

2 will be difficult :D

Reply:
Look at that when I return from the Frühlingsfest

Reply:
I used a toupe instead of a regex statement. Because it was hard to parse it back from a regex. So we should now be able to use most of the syntax that they use. What do you think?

Reply:
This will eventually break apart as well if more complex variable matches are required. FYI Regex can't be used for parsing and serialization reliably.

But, how about the following: 

```ts
type PluginSettings = {
 variableReferencePattern: {
    parse: Regex,
    serialize: (placeholder: VariableReference) => string 
  }
}

jsonPlugin({
  variableReferencePattern: {
    parse: /{}/,
    serialize: (placeholder) => `{${placeholder.name}}`
  }
})
```

Let people match with regex (because easy) but expose a callback to serialize a variable reference.

Reply:
I have the feeling that this looks pretty difficult to configure. Let's say they need to adapt it, then they have to understand the regex and the implementation to come up with something that passes a roundtrip🙈 in the end we do not now about a lot projects that use something different then {} right so why supporting every edge case from the beginning?

Reply:
Ah wait probably it's should be compatible in the future right. Okey hmmm so if we go with a solution like that, we need to provide an error handling that lets you know what went back if the parse and the serialize does not match.